cmake/ext_project_autotools.cmake

```diff
@@ -0,0 +1,118 @@
+include(utils)
+
+# Build a third-party project that uses autotools to generate build system.
+# Basically it generates configure script, run configure and then make and
+# rerun them according to their dependants in case of incremental build.
+#
+# DIR - relative path to project directory in main project tree.
+# CONFIGURE - options for configure script.
+# BYPRODUCTS - products caller targets depend upon.
+#
+# Note that incremental rebuild in case autotools files or config files are
+# changed is available only if CMake version is at least 3.12. Otherwise
+# rebuild will not be triggered. Rebuild is only required for developers
+# which presumably has newer CMake versions.
+function(ext_project_autotools name)
+    set(sv_keywords
+        DIR
+    )
+    set(mv_keywords
+        CONFIGURE
+        BYPRODUCTS
+    )
+    set(options
+        ""
+    )
+    cmake_parse_arguments(
+        ARGS "${options}" "${sv_keywords}" "${mv_keywords}" ${ARGN}
+    )
```

Comment:
1. At the first glance it was not obvious, what `sv_keywords`, `mv_keywords` and `options` mean. We can name it, say, `opts_0`, `opts_1`, `opts_multi` and I guess it will be more obvious.

    ```cmake
    # Lists of options without arguments, with one argument and
    # with multiple arguments.
    set(opts_0 "")
    set(opts_1 DIR)
    set(opts_multi CONFIGURE BYPRODUCTS)
    ```
2. Isn't `ARGS_*` variables have are too common names (can clash with others)? Shouldn't we use something from the function name in the name of the variables?

    ```cmake
    # Lists of options without arguments, with one argument and
    # with multiple arguments.
    set(epa_opts_0 "")
    set(epa_opts_1 DIR)
    set(epa_opts_multi CONFIGURE BYPRODUCTS)

    cmake_parse_arguments(
        EPA_ARGS "${epa_opts_0}" "${epa_opts_1}" "${epa_opts_multi}" ${ARGN}
    )
    ```

Reply:
https://cmake.org/cmake/help/latest/command/set.html

> Sets the given `<variable>` in the current function or directory scope.

Please, disregard the second point.

Reply:
`options`, `sv_keywords` and `mv_keywords` follow naming of `cmake_parse_arguments` arguments:

```
cmake_parse_arguments(<prefix> <options> <one_value_keywords>
                      <multi_value_keywords> <args>...)
```

`sv` and `mv` stands for single value and multiple values respectively. Also I use `set` with multiple lines formatting to be able in future extending options and keywords.

Reply:
Okay.